File: Luftborn.Server/Core/Domain/Entities/Todo.cs
using Luftborn.Server.Core.Domain.Common;
using System.ComponentModel.DataAnnotations;

namespace Luftborn.Server.Core.Domain.Entities;

public class Todo : BaseEntity
{
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(1000)]
    public string? Description { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime? DueDate { get; set; }
    public int Priority { get; set; }
} 

File: luftborn.client/src/models/Todo.js
export default class Todo {
  constructor(data = {}) {
    this.id = data.id || null;
    this.title = data.title || '';
    this.description = data.description || '';
    this.completed = data.completed !== undefined ? data.completed : false;
    this.priority = data.priority || 1; // Default priority is 1 (lowest)
  }

  static fromApi(data) {
    return new Todo({
      id: data.id,
      title: data.title,
      description: data.description,
      completed: data.isCompleted !== undefined ? data.isCompleted : data.completed,
      priority: data.priority
    });
  }

  toApiPayload() {
    return {
      title: this.title,
      description: this.description,
      completed: this.completed,
      priority: this.priority
    };
  }
} 